### ISSUE TITLE
F#: Update format specifiers for .NET versions 5 and 6

### ISSUE THREAD

**Issue by rdipardo** [ScintillaOrg/lexilla#46]
Since F# 5.0, interpolated strings accept [.NET-style specifiers] _à la_ C#, under the same conditions:

- they follow the expression, inside the braces [1]
- general form: ``[,<alignment>]:<format>[<precision>]``

   e.g.
~~~
    > $"{5./2.,-12:F4}" ;;
    val it : string = "2.5000      "
~~~

More recently, [F# 6.0] added the `%B` binary notation `printf` specifier for integers

Here's a patch and updated tests for each of the above: [LexFSharp-MoreFmtSpecs.zip](https://github.com/ScintillaOrg/lexilla/files/7737284/LexFSharp-MoreFmtSpecs.zip)

---
[1] While mixing specifier types in the same string is wrong, the lexer will highlight all recognized forms regardless; this keeps the logic simple and may help source authors identify mistakes, as the compiler's diagnostics tend to obfuscate the underlying problem:
~~~
> $"%F{5./2.,-12:F4}" ;;     

  $"%F{5./2.,-12:F4}" ;;
  ^^^^^^^^^^^^^^^^^^^

stdin(1,2): error FS3376: Invalid interpolated string. Interpolated strings may not use '%' format specifiers unless each is given an expression, e.g. '%d{1+1}'.
~~~

[.NET-style specifiers]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#standard-format-specifiers
[F# 6.0]: https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-6#formatting-for-binary-numbers

**Comment by nyamatongwe** [comment]
Committed patches.

**Comment by nyamatongwe** [comment]
Included in 5.1.5 release.
